lib/withOnyx.js

```diff
@@ -88,6 +99,14 @@ export default function (mapOnyxToState) {
              * @param {*} val
              */
             setWithOnyxState(statePropertyName, val) {
+                // This method gets called by Onyx.connect no matter if the cached
+                // data is already there or not. Thus we have to check whether
+                // the data has changed or not before we set the state.
```

Comment:
Could you please update this comment to answer the question "Why do we have to check whether the data has changed or not before setting the state?" Right now, the comment only explains what is happening and not _why_ it is happening, and I'm not entirely sure about the reason.

Reply:
I updated it, I hope its more obvious now?